UpdateFileVersionOnTag/Program.cs

```diff
@@ -0,0 +1,151 @@
+﻿using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using System.Text.RegularExpressions;
+
+
+namespace UpdateFileVersionOnTag
+{
+
+    public class Version
+    {
+        public int Major { get; private set; }
+        public int Minor { get; private set; }
+        public int Build { get; private set; }
+        public string? Label { get; private set; }
+        public int? Revision { get; private set; }
+
+        private static readonly Regex versionRegex = new Regex(@"^H2M-v(?<major>\d+)\.(?<minor>\d+)\.(?<build>\d+)(?:-(?<label>[a-zA-Z]+)(?:\.(?<revision>\d+))?)?$");
+
+        public Version(int major, int minor, int build, string? label, int? revision)
+        {
+            Major = major;
+            Minor = minor;
+            Build = build;
+            Label = label;
+            Revision = revision;
+        }
+
+        public static Version Parse(string versionString)
+        {
+            // Possible format:
+            //  H2M-v0.0.0
+            //  H2M-v0.0.0-beta
+            //  H2M-v0.0.0-beta.1
+
+            var match = versionRegex.Match(versionString);
+            if (!match.Success)
+            {
+                throw new FormatException("Version format is invalid.");
+            }
+
+            return new Version(
+                int.Parse(match.Groups["major"].Value),
+                int.Parse(match.Groups["minor"].Value),
+                int.Parse(match.Groups["build"].Value),
+                match.Groups["label"].Success ? match.Groups["label"].Value : null,
+                match.Groups["revision"].Success ? int.Parse(match.Groups["revision"].Value) : null
+            );
+
+        }
+
+        public override bool Equals(object? obj)
+        {
+            if (obj is not Version other) return false;
+
+            return Major == other.Major &&
+                   Minor == other.Minor &&
+                   Build == other.Build &&
+                   Label == other.Label &&
+                   Revision == other.Revision;
+        }
+
+        public override int GetHashCode()
+        {
+            return HashCode.Combine(Major, Minor, Build, Label, Revision);
+        }
+    }
+
+    internal class Program
+    {
+        static void Main(string[] args)
+        {
+            if (args.Length != 2)
+            {
+                throw new ArgumentException("Expected exactly two arguments: <tagVersion> <branchName>.");
+            }
+
+            string versionString = args[0];
+            string branchName = args[1];
+
+            Version version = Version.Parse(versionString);
+
+            ModifyAssemblyInfo(version);
+            ModifyLauncherService(version, branchName);
+        }
+
+        static void ModifyAssemblyInfo(Version version)
+        {
+            string h2mLauncherPath = Directory.GetParent(AppDomain.CurrentDomain.BaseDirectory).Parent.Parent.Parent.Parent.FullName;
```

Comment:
If you have a better idea to get the ``H2M-Launcher``, I take it :)

Reply:
😄 i dont know any better solution for that. Should be fine